packages/contracts/contracts/delegateTargets/VirtualAppAgreement.sol

```diff
@@ -12,47 +12,49 @@ contract VirtualAppAgreement {
   using Transfer for Transfer.Terms;
 
   struct Agreement {
-    Registry registry;
-    NonceRegistry nonceRegistry;
+    address registryAddr;
```

Comment:
I also think you can store `Registry` instead of `address` as the type

Reply:
this is passed in via calldata; are you saying to rely on some implicit conversion from `address` to `Registry`?

Reply:
yes